code/Chore.md

```diff
@@ -0,0 +1,117 @@
+# Chore
+
+Chores handle internal peer logic that needs to happen in regular intervals. Chores use databases or other services to fulfill their logic.
+
+Chores have a lifecycle, which means they must be run and shutdown.
+
+## Adding a new chore
+
+To add a new chore there are few steps:
+
+1. add a new chore implementation
+2. add a new config definition, if needed
+3. add the chore to the corresponding peer
+3.1. add it to the appropriate subsystem in the peer struct definition
+3.2. wire it together in `New`
+3.3. call `Run` in `peer.Run`
+3.3. call `Close` in `peer.Close`
+4. add config to testplanet
+
+## Implementation
+
+A basic chore implementation looks like:s
+
+```
+package orders
+
+import (
+    ...
+)
+
+// DB implements storing orders for sending to the satellite.
+type DB interface {
+	// Enqueue inserts order to the list of orders needing to be sent to the satellite.
+	Enqueue(ctx context.Context, info *Info) error
+	// ListUnsent returns orders that haven't been sent yet.
+	ListUnsent(ctx context.Context, limit int) ([]*Info, error)
+	// ListUnsentBySatellite returns orders that haven't been sent yet grouped by satellite.
+	ListUnsentBySatellite(ctx context.Context) (map[storj.NodeID][]*Info, error)
+
+	// Archive marks order as being handled.
+	Archive(ctx context.Context, satellite storj.NodeID, serial storj.SerialNumber, status Status) error
+
+	// ListArchived returns orders that have been sent.
+	ListArchived(ctx context.Context, limit int) ([]*ArchivedInfo, error)
+}
+
+// Config defines configuration for sending orders.
+type Config struct {
+	Interval time.Duration `help:"duration between sending" default:"1h0m0s"`
+	Timeout  time.Duration `help:"timeout for sending" default:"1h0m0s"`
+}
+
+// Sender sends every interval unsent orders to the satellite.
+type Sender struct {
+	// common things
+	log    *zap.Logger
+	config Config
+	// dependencies
+	transport transport.Client
+	kademlia  *kademlia.Kademlia
+	orders    DB
+
+	Loop sync2.Cycle // use cycle to run things in intervals
```

Comment:
Should this loop be accessible from outside of the package?

Reply:
Yes, for testing.